hedera-node/hedera-app/src/test/java/com/hedera/node/app/state/merkle/DependencyMigrationTest.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright (C) 2023 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.node.app.state.merkle;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+import com.hedera.hapi.node.base.SemanticVersion;
+import com.hedera.hapi.node.state.common.EntityNumber;
+import com.hedera.node.app.OrderedServiceMigrator;
+import com.hedera.node.app.ids.EntityIdService;
+import com.hedera.node.app.services.ServicesRegistryImpl;
+import com.hedera.node.app.spi.Service;
+import com.hedera.node.app.spi.fixtures.state.NoOpGenesisRecordsBuilder;
+import com.hedera.node.app.spi.info.NetworkInfo;
+import com.hedera.node.app.spi.state.MigrationContext;
+import com.hedera.node.app.spi.state.Schema;
+import com.hedera.node.app.spi.state.SchemaRegistry;
+import com.hedera.node.app.spi.state.StateDefinition;
+import com.hedera.node.app.spi.state.WritableStates;
+import com.hedera.node.app.throttle.ThrottleAccumulator;
+import com.hedera.node.config.VersionedConfigImpl;
+import com.swirlds.common.constructable.ConstructableRegistry;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import java.util.Set;
+import org.assertj.core.api.Assertions;
+import org.jetbrains.annotations.NotNull;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+@ExtendWith(MockitoExtension.class)
+class DependencyMigrationTest extends MerkleTestBase {
+    private static final long INITIAL_ENTITY_ID = 5;
+
+    @Mock
+    private VersionedConfigImpl versionedConfig;
+
+    @Mock
+    private NetworkInfo networkInfo;
+
+    private MerkleHederaState merkleTree;
+
+    @BeforeEach
+    void setUp() {
+        registry = mock(ConstructableRegistry.class);
+        merkleTree = new MerkleHederaState((tree, state) -> {}, (e, m, s) -> {}, (s, p, ds, t, dv) -> {});
+    }
+
+    @Nested
+    @SuppressWarnings("DataFlowIssue")
```

Comment:
Can't find any documentation on what this is supposed to be suppressing, and commenting it out doesn't trigger any IntelliJ inspections, so what is it?  (Oh, maybe it's Sonar ... it's still working away ...)

Reply:
ah!